File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cs
using System;
using System.Linq;

public class Solution 
{
    public string solution(string X, string Y) 
    {
        int[] countX = new int[10];
        int[] countY = new int[10];
        
        foreach (char c in X) countX[c - '0']++;
        foreach (char c in Y) countY[c - '0']++;
        
        var result = new System.Text.StringBuilder();
        
        for (int digit = 9; digit >= 0; digit--)  // 큰 숫자부터 내림차순으로
        {
            int common = Math.Min(countX[digit], countY[digit]);
            for (int i = 0; i < common; i++)
            {
                result.Append(digit);
            }
        }
        
        string str = result.ToString();
        
        if (str == "") return "-1";
        if (str[0] == '0') return "0";
        return str;
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(string X, string Y) 
{
    vector<int> Xvec(10,0);
    vector<int> Yvec(10,0);
    
    for(int i=0; i<X.length(); i++)
    {
        Xvec[X[i] - '0']++;
    }
    for(int i=0; i<Y.length(); i++)
    {
        Yvec[Y[i] - '0']++;
    }
    
    vector<int> vec;
    
    for(int i=0; i<=9; i++)
    {
        int minValue = min(Xvec[i], Yvec[i]);
        vec.push_back(minValue);
    }
    
    string result = "";
    
    for(int i=9; i>=0; i--)
    {
        if(vec[i] > 0)
        {
            for(int j=0; j<vec[i]; j++)
            {
                result += to_string(i);
            }
        }
    }
    
    bool isOverOne = false;
    
    for(int i=1; i<=9; i++)
    {
        if(vec[i] > 0) { isOverOne = true; }
    }
    
    if(!isOverOne)
    {
        if(vec[0] > 0) { result = "0"; }
        else { result = "-1"; }
    }
    return result;
}